### PR TITLE
Remove the --no-insights option and related config options

### PR DESCRIPTION
**Pull request by cnsnyder**
As we decided to go a different direction with this feature, we no longer need the --no-insights option or it's related configuration options. This reverts all the commits which collectively implemented this feature.

### CODE REVIEW

File: src/subscription_manager/managercli.py [hunk 1]
```diff
@@ -1145,10 +1144,6 @@ def _do_command(self):
                 print(_("Identity certificate has been regenerated."))
 
                 log.debug("Successfully generated a new identity from server.")
-        except connection.GoneException as ge:
```
Comment (wottop) [inline src/subscription_manager/managercli.py:1148]: Is it possible that this will still be needed?

**Review by wottop — COMMENTED** [review]
Just one question, please squash, then ack.

File: src/subscription_manager/managercli.py (hunk 1 above)
Comment (cnsnyder) [reply src/subscription_manager/managercli.py:1148]: It is, I'll add it back and squash

**Comment by wottop** [comment]
ack

**Pull request merged**